UnityProject/Assets/Scripts/Health/RespiratoryHealth/RespiratorySystem.cs

```diff
@@ -224,14 +321,19 @@ private float HandleBreathing(GasMix gasMix)
 	/// handled with interfaces we can implement so different species react differently.
 	/// </summary>
 	/// <param name="plasmaAmount"></param>
-	private void HandleBreathingPlasma(float plasmaAmount)
+	private float HandleBreathingPlasma(GasMix gasMix)
 	{
+		float plasmaPressure = gasMix.GetPressure(Gas.Plasma);
+
+		float plasmaConsumed = 0;
+
+		plasmaConsumed = gasMix.GetMoles(Gas.Plasma) * AtmosConstants.BREATH_VOLUME;
```

Comment:
If I'm understanding what you're doing here correctly it would most likely be cleaner to make the input gasmix scale, rather then doing it each time you want to consume a gas.

Reply:
this is simply a copy of what oxygen inhalation does. it takes the current amount of the gas and multiplies it by how big your lungs are. it does this for both oxygen and plasma.

Reply:
I'll admit it's a bit hard to tell since I don't know the codebase, but `HandleBreathingPlasma` expects the gasmix we're breathing from right? Like the tile or gas tank? Why not scale that gasmix before passing it in, so we can treat it as the whole "breath" without needing to worry about scaling it?

Reply:
mainly due to the pressure thing. the "pressure" part is necessary to figure out whether you should die or not, hence it should calculate from the tile or tank's pressure. however, the amount consumed by breathing needs to be scaled.

Reply:
Ah I see you're handling the damage to the body/lungs in the same bit as the breathing.

Reply:
The reason this is so... unrefined is because this is largely a placeholder until actual body parts are implemented. For now this will have to do.

Reply:
F